helloworld/front/react/src/App.tsx

```diff
@@ -0,0 +1,70 @@
+import "./App.css";
+import "@massalabs/react-ui-kit/src/global.css";
+import { useEffect, useState } from "react";
+import { IAccount, providers } from "@massalabs/wallet-provider"
+import { ClientFactory } from "@massalabs/massa-web3";
+import { Args } from "@massalabs/massa-web3";
+
+const CONTRACT_ADDRESS = "AS1u8i5H1RQU5qD8R8hQzugA8HwWmS9qqyZNjhvR9WywUP17v1od";
+
+function App() {
+    const [errorMessage, setErrorMessage] = useState<any>("");
+    const [nodeUrls, setNodeUrls] = useState<string[] | null>(null);
+    const [account, setAccount] = useState<IAccount | null>(null);
+    const [lastOpId, setlastOpId] = useState<string | null>(null);
+
+    useEffect(() => {
+        const registerAndSetProvider = async () => {
+        try {
+           let provider = (await providers(true, 10000))[0];
+           let accounts = await provider.accounts();
+            if (accounts.length == 0) {
+                setErrorMessage("No accounts found");
+                return;
+            }
+            setAccount(accounts[0]);
+            setNodeUrls(await provider.getNodesUrls());
+        } catch (e) {
+            console.log(e);
+            setErrorMessage("Please install massa station and the wallet plugin of Massa Labs and refresh.");
+        }
+        };
+
+        registerAndSetProvider();
+    }, []);
+
+    const callHelloWorld = async () => {
+        try {
+            if (!nodeUrls || !account) {
+                return;
+            }
+            let client = await ClientFactory.createClientFromWalletProvider(nodeUrls, account);
```

Comment:
Why do you need to pass something to the walletProvider ? You should be able to directly call fromWalletProvider without arguments and let this function call wallet provider to extract the needed info.
And yes, I renamed the function as ClientFactory.fromWalletProvider is self explanatory I guess :-)

Reply:
Done :)

Reply:
fixed on massaweb3:  nodeUrls replaced by provider. The rest of the simplificatino is to be done at a component level.